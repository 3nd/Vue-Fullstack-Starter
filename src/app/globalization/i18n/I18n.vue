<template>
  <v-layout>
    <v-btn @click="setLanguage('en')">English</v-btn>
    <v-btn @click="setLanguage('zh')">中文</v-btn>
    <v-btn @click="setLanguage('ja')">日本語</v-btn>
    <p>{{ $t('message.hello') }}</p>
  </v-layout>
</template>

<script>
export default {};
</script>

<style scoped>
</style>
