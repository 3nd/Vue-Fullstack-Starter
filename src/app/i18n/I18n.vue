<i18n src="./translation.yml"></i18n>

<template>
  <v-layout column>

    <v-layout row>
      <div>
        <v-btn @click="setLanguage('en')">English</v-btn>
        <v-btn @click="setLanguage('zh')">中文</v-btn>
        <v-btn @click="setLanguage('ja')">日本語</v-btn>
      </div>
    </v-layout>

    <v-layout row>
      <p class="pl-2 pt-4">
        HTML formatting:
        <b class="indigo--text">{{ $t('bubble') }}</b>
      </p>
    </v-layout>

    <v-layout row>
      <p class="pl-2">
        Named formatting:
        <b class="indigo--text">{{ $t('hello', { msg: 'Vue' }) }}</b>
      </p>
    </v-layout>

    <v-layout row>
      <p class="pl-2">
        List formatting:
        <b class="indigo--text">{{ $t('thinkIn', ['Angular', 'React', 'Vue']) }}</b>
      </p>
    </v-layout>

    <v-layout row>
      <p class="pl-2">
        Pluralization:
        <b class="indigo--text">{{ $tc('http', 2) }}://google.com</b>
      </p>
    </v-layout>

    <v-layout row>
      <p class="pl-2">
        Pluralizing i18n:
        <b class="indigo--text">{{ $tc('framework', 0) }}</b>,
        <b class="indigo--text">{{ $tc('framework', 1) }}</b>,
        <b class="indigo--text">{{ $tc('framework', 3, { num: 3 }) }}</b>
      </p>
    </v-layout>

  </v-layout>
</template>

<script>
export default {
  created() {
    const languages = ['en', 'zh', 'ja'];
    languages.forEach((item) => {
      if ((navigator.language).includes(item)) {
        this.$i18n.locale = item;
      }
    });
  },
  methods: {
    setLanguage(val) {
      this.$i18n.locale = val;
    },
  },
};
</script>

<style scoped>
</style>
