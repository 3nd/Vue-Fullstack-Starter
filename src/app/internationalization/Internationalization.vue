<i18n src="./translation.yml"></i18n>

<template>
  <v-layout column>
    <div class="headline">Internationalization</div>

    <v-layout row>
      <p class="pl-2 pt-4">
        HTML formatting:
        <b>{{ $t('bubble') }}</b>
      </p>
    </v-layout>

    <v-layout row>
      <p class="pl-2">
        Named formatting:
        <b>{{ $t('hello', { msg: 'Vue' }) }}</b>
      </p>
    </v-layout>

    <v-layout row>
      <p class="pl-2">
        List formatting:
        <b>{{ $t('thinkIn', ['Angular', 'React', 'Vue']) }}</b>
      </p>
    </v-layout>

    <v-layout row>
      <p class="pl-2">
        Pluralization:
        <b>{{ $tc('http', 2) }}://google.com</b>
      </p>
    </v-layout>

    <v-layout row>
      <p class="pl-2">
        Pluralizing i18n:
        <b>{{ $tc('framework', 0) }}</b>,
        <b>{{ $tc('framework', 1) }}</b>,
        <b>{{ $tc('framework', 3, { num: 3 }) }}</b>
      </p>
    </v-layout>

  </v-layout>
</template>

<script>
export default {};
</script>

<style scoped>
</style>
