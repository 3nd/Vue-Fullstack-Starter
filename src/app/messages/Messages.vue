<template>
  <v-container fluid>
    <div class="headline">Messages</div>
  </v-container>
</template>

<script>
// @flow

export default {};
</script>

<style scoped>
</style>
