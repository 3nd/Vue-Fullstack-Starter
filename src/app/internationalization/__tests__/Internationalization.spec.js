import { createLocalVue, shallow } from '@vue/test-utils';
import I18n from 'vue-i18n';

import Internationalization from '../Internationalization';
import messages from '../_languages/en.yml';

const localVue = createLocalVue();

localVue.use(I18n);

describe('Internationalization', () => {
  let [wrapper, i18n] = [];

  beforeEach(() => {
    i18n = new I18n({
      locale: 'en',
      messages,
    });

    wrapper = shallow(Internationalization, { localVue, i18n });
  });

  it('should render initial component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
