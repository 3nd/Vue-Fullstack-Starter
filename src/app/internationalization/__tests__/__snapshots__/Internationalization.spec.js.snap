// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Internationalization should render initial component 1`] = `
<v-layout column="">
  <div class="headline">Internationalization</div>
  <v-layout row="">
    <p class="pl-2 pt-4">
      HTML formatting:
      <b>Bubble</b>
    </p>
  </v-layout>
  <v-layout row="">
    <p class="pl-2">
      Named formatting:
      <b>Hello, Vue!</b>
    </p>
  </v-layout>
  <v-layout row="">
    <p class="pl-2">
      List formatting:
      <b>Think in Vue</b>
    </p>
  </v-layout>
  <v-layout row="">
    <p class="pl-2">
      Pluralization:
      <b>https://google.com</b>
    </p>
  </v-layout>
  <v-layout row="">
    <p class="pl-2">
      Pluralizing i18n:
      <b>no framework</b>,
      <b>one framework</b>,
      <b>3 frameworks</b>
    </p>
  </v-layout>
</v-layout>
`;
